### ISSUE TITLE
If Mayor Noggenfogger directs a spell to Primalfin Champion, its deathrattle will return the spell to your hand without "Inspired" enchantment

### ISSUE THREAD

**Issue by VoidMayonnaise** [HearthSim/hs-bugs#737]
https://youtu.be/iuPimqpINVs
And I have some other test results about Primalfin Champion (PC).
①PC+Faceless Manipulator: FM will perfectly copy the PC. Its deathrattle can return the spells cast on the former PC as usual.
②PC+Unearthed Raptor: UR will only copy the deathrattle. You won't get the spells cast on the PC, but UR can also gain "Inspired" and its deathrattle can also return the spells as usual if you cast spells on it.
③PC+Counterspell/Spellbender (cast a spell on PC): PC will gain "Inspired" enchantment but its deathrattle won't return the spell.
④PC+Tortollan Primalist (cast on PC): PC won't gain "Inspired" enchantment, and its deathrattle won't return the spells either (maybe it's only because this kind of spell is special).
When I try to explain ① and ②, I suppose that the deathrattle will read the enchantment list of the minion, and this perfectly explained why UR can't return the spells cast on PC (since UR can only copy the deathrattle but it can't copy enchantments). However, this one is the exact opposite of ③ and this issue, which show that "Inspired" enchantment seems to be unrelated with the deathrattle.
So this is my opinion:
PC will gain "Inspired" enchantment during Play Phase (but it's useless). During the Spell Text Phase, the spell is recorded by another way (may be similar to Moat Lurker). When the deathrattle triggers, it will read the list of recorded spells and add them to the player's hand.
But the problem is: why the "Inspired" enchantment is useless?

**Comment by Patashu** [comment]
'Inspired' Enchantment may do nothing. It may only exist to visually display to the user that they should expect to get spells back.

**Comment by VoidMayonnaise** [comment]
If so,  I think that there should be a bug between PC and Unearthed Raptor.  Since the spells aren't stored by enchantment list, Unearthed Raptor should also copy the stored spell list, just like Moat Lurker.

**Comment by Patashu** [comment]
Yeah, it does seem inconsistent with Moat Lurker/Unearthed Raptor...
